src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -181,26 +182,25 @@ internal static bool DoSslHandshake(SafeSslHandle context, byte[] recvBuf, int r
             return stateOk;
         }
 
-        internal static int Encrypt(SafeSslHandle context, byte[] input, int offset, int count, ref byte[] output, out Ssl.SslErrorCode errorCode)
+        internal static int Encrypt(SafeSslHandle context, ReadOnlyMemory<byte> input, ref byte[] output, out Ssl.SslErrorCode errorCode)
         {
-            Debug.Assert(input != null);
-            Debug.Assert(offset >= 0);
-            Debug.Assert(count > 0);
-            Debug.Assert(offset <= input.Length);
-            Debug.Assert(input.Length - offset >= count);
-
             errorCode = Ssl.SslErrorCode.SSL_ERROR_NONE;
 
             int retVal;
             unsafe
             {
-                fixed (byte* fixedBuffer = input)
+                MemoryHandle handle = input.Retain(true);
+                try
                 {
-                    retVal = Ssl.SslWrite(context, fixedBuffer + offset, count);
+                    retVal = Ssl.SslWrite(context, (byte*)handle.PinnedPointer, input.Length);
+                }
+                finally
+                {
+                    handle.Dispose();
```

Comment:
Nit: this can be:
```C#
using (MemoryHandle handle = input.Retain(pin: true))
{
    retVal = Ssl.SslWrite(context, (byte*)handle.PinnedPointer, input.Length);
}
```

Reply:
yeah, my VS was being silly and said I couldn't ..  but a fresh install says all good, so fixed.